avail.go

```diff
@@ -0,0 +1,156 @@
+package avail
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/rollkit/go-da"
+)
+
+type SubmitRequest struct {
+	Data string `json:"data"`
+}
+
+type SubmitResponse struct {
+	BlockNumber      uint32 `json:"block_number"`
+	BlockHash        string `json:"block_hash"`
+	TransactionHash  string `json:"hash"`
+	TransactionIndex uint32 `json:"index"`
+}
+
+type BlocksResponse struct {
+	BlockNumber      uint32             `json:"block_number"`
+	DataTransactions []DataTransactions `json:"data_transactions"`
+}
+
+type DataTransactions struct {
+	Data      string `json:"data"`
+	Extrinsic string `json:"extrinsic"`
+}
+
+type Config struct {
+	LcURL     string `json:"lc_url"`
+	BlocksURL string `json:"blocks_url"`
+}
+
+// AvailDA implements the avail backend for the DA interface
+type AvailDA struct {
+	appID  uint32
+	config Config
+	ctx    context.Context
+}
+
+// NewAvailDA returns an instance of AvailDA
+func NewAvailDA(appID uint32, ctx context.Context) *AvailDA {
+	return &AvailDA{
+		appID:  appID,
+		ctx:    ctx,
+		config: Config{LcURL: "http://localhost:8000/v2", BlocksURL: "/blocks/%d/data?fields=data,extrinsic"},
```

Comment:
Can we make this configurable, e.g. with CLI argument or configuration file? Only Light Client URL, I guess that BlocksURL can be constant.